libraries/shared/watcher/storage_watcher.go

```diff
@@ -89,31 +91,27 @@ func (watcher StorageWatcher) Execute() error {
 }
 
 func (watcher StorageWatcher) transformDiffs() error {
-	diffs := make(chan types.PersistedDiff)
-	errs := make(chan error)
-	done := make(chan bool)
-
-	defer close(diffs)
-	defer close(errs)
-	defer close(done)
-
-	go watcher.StorageDiffRepository.GetNewDiffs(diffs, errs, done)
-
+	minID := 0
 	for {
-		select {
-		case diff := <-diffs:
-			err := watcher.transformDiff(diff)
-			if err != nil {
-				if err == sql.ErrNoRows || reflect.TypeOf(err) == reflect.TypeOf(types.ErrKeyNotFound{}) {
-					logrus.Tracef("error transforming diff: %s", err.Error())
+		diffs, extractErr := watcher.StorageDiffRepository.GetNewDiffs(minID, ResultsLimit)
+		if extractErr != nil {
+			return fmt.Errorf("error getting unchecked diffs: %s", extractErr.Error())
+		}
+		for _, diff := range diffs {
+			transformErr := watcher.transformDiff(diff)
+			if transformErr != nil {
+				if transformErr == sql.ErrNoRows || reflect.TypeOf(transformErr) == reflect.TypeOf(types.ErrKeyNotFound{}) {
+					logrus.Tracef("error transforming diff: %s", transformErr.Error())
 				} else {
-					logrus.Infof("error transforming diff: %s", err.Error())
+					logrus.Infof("error transforming diff: %s", transformErr.Error())
```

Comment:
I think we need to think about cases where we should potentially return this error and restart the service (e.g. if we lose the database connection) but didn't want to bloat this diff